src/deepsparse/transformers/pipelines/zero_shot_text_classification.py

```diff
@@ -0,0 +1,334 @@
+    Each sequence and each candidate label must be paired and passed through
+    the model, so the total number of forward passes is num_labels * num_sequences
+    """
+
+    sequences: Union[List[List[str]], List[str], str] = Field(
+        description="A string or List of strings representing input to "
+        "zero_shot_text_classification task"
+    )
+
+
+class ZeroShotTextClassificationOutput(BaseModel):
+    """
+    Schema for zero_shot_text_classification pipeline output. Values are in batch order
+    """
+
+    sequences: Union[List[List[str]], List[str], str] = Field(
+        description="A string or List of strings representing input to "
+        "zero_shot_text_classification task"
+    )
+    labels: Union[List[List[str]], List[str]] = Field(
+        description="The predicted labels in batch order"
+    )
+    scores: Union[List[List[float]], List[float]] = Field(
+        description="The corresponding probability for each label in the batch"
+    )
+
+
+class ZeroShotTextClassificationImplementation(TransformersPipeline):
+    """
+    Base class for implementing zero shot text classification. Implementations of
+    zero shot text classification inherit from this class.
+
+    :param model_path: sparsezoo stub to a transformers model, an ONNX file, or
+        (preferred) a directory containing a model.onnx, tokenizer config, and model
+        config. If no tokenizer and/or model config(s) are found, then they will be
+        loaded from huggingface transformers using the `default_model_name` key
+    :param engine_type: inference engine to use. Currently supported values include
+        'deepsparse' and 'onnxruntime'. Default is 'deepsparse'
+    :param num_cores: number of CPU cores to allocate for inference engine. None
+        specifies all available cores. Default is None
+    :param scheduler: (deepsparse only) kind of scheduler to execute with.
+        Pass None for the default
+    :param input_shapes: list of shapes to set ONNX the inputs to. Pass None
+        to use model as-is. Default is None
+    :param alias: optional name to give this pipeline instance, useful when
+        inferencing with multiple models. Default is None
+    :param sequence_length: sequence length to compile model and tokenizer for.
+        If a list of lengths is provided, then for each length, a model and
+        tokenizer will be compiled capable of handling that sequence length
+        (also known as a bucket). Default is 128
+    :param default_model_name: huggingface transformers model name to use to
+        load a tokenizer and model config when none are provided in the `model_path`.
+        Default is "bert-base-uncased"
+    :param context: context for engine. If None, then the engine will be initialized
+        with 2 streams to make use of parallel inference of labels
+    """
+
+    def __init__(
+        self,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+
+    def parse_config(
```

Comment:
let's move parse_config and parse_labels to bottom of this class' functions to allow for base `Pipeline` overrides to come first